packages/babel-plugin-transform-runtime/src/index.js

```diff
@@ -254,7 +254,14 @@ export default declare((api, options, dirname) => {
         if (cached) {
           cached = t.cloneNode(cached);
         } else {
-          cached = addDefault(file.path, source, {
+          const moduleTransforms = file.get(
+            "@babel/plugin-transform-modules-*",
+          );
+          // Remove the trailing `.js` requires by ES Module when module transforms presents
+          const validSource = moduleTransforms
```

Comment:
Technically only `amd` and `umd` are affected by this issue since by default RequireJS does not support resolving `foo.js => foo.js` as-is out of the box. You may configure to support this but we did cause breaking changes to these users.

Both CommonJS and SystemJS support `foo.js => foo.js`, but at the same time they also support `foo => foo.js`. Hence I remove `.js` for the sake of briefness. C.f. https://github.com/babel/babel/pull/10549#pullrequestreview-306294014